frontend/src/utils/handleCaptionsSignal/handleCaptionsSignal.ts

```diff
@@ -0,0 +1,63 @@
+import { Dispatch, SetStateAction } from 'react';
+import { SignalEvent } from '../../types/session';
+
+/**
+ * @typedef {object} CaptionsSignalDataType
+ * @property {string} action - The action to be performed on captions (e.g., 'enable', 'disable', 'join', 'leave').
+ * @property {string} captionsId - The ID of the captions to be enabled or disabled.
+ */
+export type CaptionsSignalDataType = {
+  action: 'enable' | 'disable';
+  captionsId: string;
+};
+
+/**
+ * @typedef {object} CaptionsHandleType
+ * @property {SignalEvent} event - The signal event containing the data.
+ * @property {Dispatch<SetStateAction<boolean>>} setIsCaptioningEnabled - Function to set the captions enabled state.
+ */
+export type CaptionsHandleType = {
+  event: SignalEvent;
+  setIsCaptioningEnabled: Dispatch<SetStateAction<boolean>>;
+};
+
+/**
+ * Handles the captions signal received from the session.
+ * This function manages enabling, disabling, joining, and leaving captions.
+ * It also handles the functionality of notifying other participants about the current captions ID and the participant count.
+ * @param {CaptionsHandleType} props - Dependencies needed for the handler
+ * @property {SignalEvent} event - The signal event containing the data.
+ * @property {Dispatch<SetStateAction<boolean>>} setIsCaptioningEnabled - Function to set the captions enabled state.
+ */
+const handleCaptionsSignal = ({ event, setIsCaptioningEnabled }: CaptionsHandleType) => {
+  try {
+    if (!event.data) {
+      return;
+    }
+
+    const parsedData: CaptionsSignalDataType = JSON.parse(event.data);
+    const { action, captionsId } = parsedData;
+
+    switch (action) {
+      case 'enable':
+        if (captionsId) {
+          setIsCaptioningEnabled(true);
+        }
+        break;
```

Comment:
In addition to this hook, could we add functionality for the room route on the backend to set the state? As-is, if a user joins a room after captions have been enabled, they won't be aware they're being captioned (the button is greyed out). The captions box also doesn't show up, either.

Reply:
synced up w/ Chris on this one offline, we decided to keep the current flow as-is for the time being at least: if one user enables captions for themselves locally, while it is enabled session-wide, a different user will have to click it themselves to see the captions.